### PR TITLE
Get remote source components of source container

### PR DESCRIPTION
**Pull request by hanchuntao**
[OLCS-246](https://issues.redhat.com/browse/OLCS-246)

Fixes OLCS-246

### CODE REVIEW

**Comment by yulwang0623** [comment]
Fixes OLCS-246

**Comment by yulwang0623** [comment]
Fixes OLCS-246

**Comment by yulwang0623** [comment]
Fixes OLCS-246

**Comment by hanchuntao** [comment]
@rhyw @qduanmu @huiwangredhat PTAL

File: openlcs/libs/kojiconnector.py [hunk 1]
```diff
@@ -325,5 +329,67 @@ def download_container_image_archives(self, build, dest_dir, arch=None):
...
-                raise ValueError(err_msg)
+                raise ValueError("No images found.") from None
+
+    @staticmethod
+    def get_component_flat(data):
+        return {
+            "uuid": str(uuid.uuid4()),
+            "type": data.get("type"),
+            "name": data.get("name"),
+            "version": data.get("name", ""),
```
[the remark this answers is not in the document]
Comment (hanchuntao) [reply openlcs/libs/kojiconnector.py]: This is a mistake, should be `version`, updated

File: openlcs/libs/kojiconnector.py [hunk 2]
```diff
@@ -325,5 +329,72 @@ def download_container_image_archives(self, build, dest_dir, arch=None):
...
+            "version": data.get("version", ""),
+            "release": "",
+            "license": "",
+            "arch": "",
+            'synced': False
+        }
+
+    def parse_remote_source_components(self, rs_archive_url):
+        """
+        Parse components from remote source archive url.
```
Comment (huiwangredhat) [inline openlcs/libs/kojiconnector.py:349]: I guess the rs_archive_url should be the binary container's remote-source.json, not the remote-source.tar.gz. So would you like to give more description?

File: openlcsd/flow/tasks.py [hunk 3]
```diff
@@ -751,6 +750,40 @@ def get_components_product_from_corgi(context, engine):
...
+    config = context.get('config')
+    sc_nvr = context.get('package_nvr')
+    # Get components from the source container itself.
+    srpm_dir = context.get('srpm_dir')
+    misc_dir = context.get('misc_dir')
+    sc_handler = SourceContainerHandler(config)
+    components = sc_handler.get_container_components(srpm_dir, misc_dir,
+                                                     sc_nvr)
+    # Get remote source components from its binary container.
+    components.update(get_remote_source_components(context, engine))
```
Comment (huiwangredhat) [inline openlcsd/flow/tasks.py:782]: It's ok for me to update the components. Getting remote source components to the sc_handler.get_container_components is more consistent.

File: openlcs/libs/kojiconnector.py (hunk 2 above)
Comment (hanchuntao) [reply openlcs/libs/kojiconnector.py:349]: Please check the function `get_remote_source_components` in `kojiconnector.py`, this `rs_archive_url ` is the url for `remote-source.json`

File: openlcsd/flow/tasks.py (hunk 3 above)
Comment (hanchuntao) [reply openlcsd/flow/tasks.py:782]: They use different functions, srpms use function in sc_handler, but for remote source, get component form koji, so use the function defined in `kojiconnector.py`

**Comment by huiwangredhat** [comment]
LGTM

**Pull request merged**